backfiller/backfiller/main.py

```diff
@@ -258,8 +257,8 @@ def delete_hours(self):
 		else:
 			self.logger.info('Deleting hours older than {} hours ago'.format(self.start))
 
-		for quality in self.qualities:
-			hours = list_local_hours(self.base_dir, self.channel, quality)
+		for channel, quality in itertools.product(self.channels, self.qualities):
```

Comment:
I guess this is more efficient if there a bunch of channels that don't have many segments than spawning a manager for each.